docs/DeveloperGuide.md

```diff
@@ -5,9 +5,238 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
+### Adding calorie entries feature
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Implementation
 
+This functionality is facilitated by `UI`, `CalorieList`, `FileHandler` and `ParserCalories`. It implements one operation, namely:
+- `UI#handleCaloriesInput(String, CalorieList)`
+- `CalorieList#addEntry(String)`
+- `ParserCalories#parseCaloriesInput(String)`
+- `FileHandler#updateFile()`
+
+This feature is activated when the user inputs a `calories in` or `calories out` command in the terminal.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+
+- Step 1: When the user inputs the command `calories in burger c/200 d/2024-02-02` in the terminal,
+the string is sent to `UI#handleCaloriesInput(String, CalorieList)`, which calls `CalorieList#addEntry(String)`.
+
+- Step 2: Inside `CalorieList#addEntry(String)`, the function `ParserCalories#parseCaloriesInput(String)` is then called to extract information such as the description, number of calories, and date of entry.
+
+- Step 3: The obtained information is sent to the private method `ParserCalories#makeNewInputEntry(String, int, String)` to create a new entry of class `InputEntry` that extends `Entry`. An `Entry` object is then returned to the caller, `CalorieList#addEntry(String)`.
+
+- Step 4: The returned `Entry` object is added into the `calorieArrayList` member of type `ArrayList<Entry>` in the `CalorieList`, via the `ArrayList.add()` method.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `CalorieList`.
+
+The sequence diagram for this feature is shown below:
+
+![CaloriesAddEntrySeqDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesAddEntrySeqDiagram.puml)
```

Comment:
As a peer mentioned, you are missing the colon.
The actor `Bob` is a good adition, but is `Bob` really a good way of describing a `User`?